aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1164,18 +1158,32 @@ AT_ERROR("solve: MAGMA library not found in "
 std::tuple<Tensor, Tensor> _solve_helper_cuda(const Tensor& self, const Tensor& A) {
   auto self_working_copy = cloneBatchedColumnMajor(self);
   auto A_working_copy = cloneBatchedColumnMajor(A);
-  std::vector<int64_t> infos(batchCount(self), 0);
+  auto infos = at::empty({std::max<int64_t>(1, batchCount(self))}, self.options().dtype(kInt));
```

Comment:
So is this pretending that `apply_solve` can use `infos` on CUDA? 

What if we just create `infos` on the CPU to begin with?

Reply:
For CUDA there are two different MAGMA calls `magmaSolve`([`magma_*gesv_gpu`](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__gesv.html)) and `magmaSolveBatched`([`magma_*gesv_batched`](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__gesv__batched.html)).
For single matrix solves MAGMA requires a pointer to an integer residing in CPU memory, while for batched input MAGMA requires a pointer to an array of integers residing in device memory. I consider the batched case to be more used and therefore `infos` should be created on the GPU by default.
So `apply_solve` does use `infos` directly on CUDA and without movement of data between devices (at least visible to us, there could be some sync inside the magma function) but only for the batched case.
I agree that we should probably create `infos` on the CPU for the single matrix solves now.
But in general for all other future linear algebra functions `infos` can and should be on GPU, but it depends on the particular calls to the underlying library.

Reply:
Interesting. We'll have a chance to revisit this when we look at cross-device synchronization more closely. cc @heitorschueroff. This seems OK for now. 

Thanks for your detailed explanation, @IvanYashchuk.